x/bridge/keeper/msg_server.go

```diff
@@ -0,0 +1,133 @@
+// Package keeper TODO: upgrade the signatures validation process
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/osmosis-labs/osmosis/v23/x/bridge/types"
+)
+
+var _ types.MsgServer = msgServer{}
+
+type msgServer struct {
+	k Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{k: keeper}
+}
+
+func (m msgServer) InboundTransfer(
+	goCtx context.Context,
+	msg *types.MsgInboundTransfer,
+) (*types.MsgInboundTransferResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if !m.k.validateSender(ctx, msg.Sender) {
+		return nil, errorsmod.Wrapf(sdkerrors.ErrorInvalidSigner, "Sender is not part of the signer set")
+	}
+
+	err := m.k.InboundTransfer(ctx, msg.DestAddr, msg.Asset, msg.Amount)
+	if err != nil {
+		return nil, err
+	}
+	err = ctx.EventManager().EmitTypedEvent(&types.EventInboundTransfer{
+		Sender:   msg.Sender,
+		DestAddr: msg.DestAddr,
+		Asset:    msg.Asset,
+		Amount:   msg.Amount,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return new(types.MsgInboundTransferResponse), nil
+}
+
+func (m msgServer) OutboundTransfer(
+	goCtx context.Context,
+	msg *types.MsgOutboundTransfer,
+) (*types.MsgOutboundTransferResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Don't need to check the signature here since every user could be the sender
+	// TODO: DestAddr must be a part of the TSS valset?
```

Comment:
Can you explain what this TODO means? Shouldn't the DestAddr just be any valid address on the destination chain and have nothing to do with TSS valset?

Reply:
You are right, my fault. Something mixed up in my head when I was writing this code :) 
I deleted this todo